### DISCUSSION TITLE
Jinja2 is breaking when it uses MarkeupSafe 2.1.0.

### DISCUSSION THREAD

**Discussion by thompsonp17** [pallets/jinja#1602] [Q&A] [answered]
Folks, there is an error raising from Jinja2 when it refers to **soft_unicode** from MarkeupSafe. I've found that safe_unicode was replaced in version 2.1.0.

Regards

**Comment by ThiefMaster** [comment]
**Please** read all the existing issues about pinning your transitive dependencies. There are literally tens of issues about this.

**Comment by davidism** [comment, marked as answer]
You are using an unsupported version of Jinja, please update to the latest version if possible. Additionally, please read https://hynek.me/articles/semver-will-not-save-you/, then use a tool like [pip-tools](https://pypi.org/project/pip-tools/) to pin your dependencies and control when you get updates. Be sure to run your tests with deprecation warnings treated as errors so that you get notified of these types of changes early.
